src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -1902,6 +1906,23 @@ else if (unhandledUnerased.nonEmpty()) {
         }
     }
     // where
+        private boolean shouldCheckPreview(MethodSymbol m, MethodSymbol other, ClassSymbol origin) {
+            if (m.owner != origin ||
+                //performance - only do the expensive checks when the overridden method is a Preview API:
+                (other.flags() & PREVIEW_API) == 0) {
```

Comment:
indentation seems off here - e.g. I'd expect double "tab" in the second line of the `if` condition?